src/game/mario_actions_moving.c

```diff
@@ -435,6 +435,19 @@ void update_walking_speed(struct MarioState *m) {
         // Slow down if moving backwards
         m->forwardVel += 1.1f;
     } else if (m->forwardVel <= targetSpeed) {
+        
+        // When starting a walk, make a few checks and set Mario's speed to 8 on the first frame.
+        // This ensures Mario's speed is set consistently when starting a walk.
+        // We use m->actionArg here since it's a global variable that gets initalized to 0 elsewhere.
+        if (m->forwardVel <= 8.f && !m->actionArg
```

Comment:
Probably makes more sense to use `actionTimer` for this instead of `actionArg`. `actionTimer` would need to be incremented somewhere (probably this function) rather than hardset to 1, and would need a conditional to cap it.

This is still a movement change so I'm hesitant to merge this without some sort of define for this, even if it's ultimately not inside a config file. Seems like this isn't even useful for framewalking thanks to the steep slope check, so idk how much benefit this offers anyway (Kaze seems to think it's noticeable though).

Reply:
WE ARE NOT MAKING THIS A DEFINE!!!!!!

Reply:
Hey, I did say it'd need to be housed outside of the config file...